Mobile/lib/Components/TaskEdit.dart

```diff
@@ -0,0 +1,281 @@
+import 'package:flutter/material.dart';
+import 'package:mobile/Components/SelectDateTime.dart';
+import 'package:mobile/config/app_pages.dart';
+import 'package:mobile/models/task.dart';
+import 'package:mobile/services/app_state.dart';
+import 'package:provider/provider.dart';
+
+class TaskEdit extends StatefulWidget {
+  final Task task;
+  
+  const TaskEdit({
+    super.key,
+    required this.task
+  });
+
+  @override
+  State<TaskEdit> createState() => _TaskEditState();
+}
+
+class _TaskEditState extends State<TaskEdit> {
+  final _formKey = GlobalKey<FormState>();
+  final Selectdatetime selectdatetime = new Selectdatetime();
+  final TextEditingController _endDatetimestampController = TextEditingController();
+  late RootAppState _appState;
+  
+  late Task oldTask;
+  late String title;
+  late String description;
+  late String reward;
+  late DateTime endDate;
+  late bool recurring;
+  late String recurringInterval;
+  late bool singleCompletion;
+  
+  @override
+  void initState() {
+    super.initState();
+    oldTask = widget.task;
+    title = oldTask.title;
+    description = oldTask.description;
+    reward = oldTask.reward.toString();
+    endDate = oldTask.endDate;
+    recurring = oldTask.recurring;
+    recurringInterval = oldTask.recurringInterval.toString();
+    singleCompletion = oldTask.singleCompletion;
+
+    _endDatetimestampController.text = endDate.toString();
+  }
+
+  void _deleteTaskConfimation() {
+    showDialog(
+      context: context,
+      builder: (context) => AlertDialog(
+        title: Text('Delete task: ' + oldTask.title),
+        actions: [
+          TextButton(
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+            child: Text('Close'),
+          ),
+          TextButton(
+            onPressed: _deleteTask,
+            child: Text('Confirm'),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _deleteTask() async {
+    int statusCode = await _appState.deleteTask(oldTask.id);
+    if (statusCode == 204) {
+      Navigator.of(context).pop();
+      Navigator.of(context).pop();
+      Navigator.of(context).pop();
+      _appState.switchPage(_appState.page);
```

Comment:
This line seems redundant, as you're asking it to switch to current page.
```suggestion
```

Reply:
tried to reload the side... didn't work and forgot to remove